### PR TITLE
Encode url before handing to RestClient to execute a request

### PR DESCRIPTION
**Pull request by bzwei**
Fixes issue #183

### CODE REVIEW

File: lib/azure/armrest/armrest_service.rb [hunk 1]
```diff
@@ -161,7 +161,11 @@ class << self
         private
 
         def rest_execute(options, http_method = :get)
-          RestClient::Request.execute(options.merge(:method => http_method))
+          options = options.merge(
+            :method => http_method,
+            :url    => Addressable::URI.parse(options[:url]).normalize.to_str
```
Comment (djberg96) [inline lib/azure/armrest/armrest_service.rb]: What does .parse + .normalize + .to_str do that .escape doesn't do?

Comment (bzwei) [inline lib/azure/armrest/armrest_service.rb]: @djberg96 I actually don't know. I just followed what is said in  http://stackoverflow.com/questions/20254749/restclient-parsing-uri-with-space-on-request-path that "The documentation suggests to parse and normalize the URL first".
I don't know what document it referenced 😄

Comment (djberg96) [inline lib/azure/armrest/armrest_service.rb]: Well, I'd rather not call extra methods if we don't have to. Could we add some tests?

**Comment by djberg96** [comment]
👍

**Pull request merged**